dashboard/test/models/user_test.rb

```diff
@@ -13,6 +12,36 @@ class UserTest < ActiveSupport::TestCase
     assert_equal '21+', teacher.age
   end
 
+  # Disable this test if and when we do require teachers to complete school data
+  test 'school info should not be validated' do
+    school_attributes = {
+      country: 'US',
+      school_type: SchoolInfo::SCHOOL_TYPE_PUBLIC,
+      state: nil
+    }
+    teacher = create :teacher, school_info_attributes: school_attributes
+    teacher.save!
+    assert teacher.valid?, teacher.errors.full_messages
+  end
+
+  test 'identical school info should not be duplicated in the database' do
+    school_attributes = {
+      country: 'US',
+      school_type: SchoolInfo::SCHOOL_TYPE_PUBLIC,
+      state: 'CA'
+    }
+    teacher = create :teacher, school_info_attributes: school_attributes
+    teacher.save!
+
+    teacher2 = create :teacher, school_info_attributes: school_attributes
+    teacher2.save!
+
+    attr = teacher.process_school_info_attributes(school_attributes)
+    school_info = SchoolInfo.where(attr).first
+    assert teacher.school_info == school_info && teacher2.school_info == school_info
+    assert SchoolInfo.where(attr).count == 1
```

Comment:
TIL: The `2.times` loop can be simplified to `create_list(:teacher, 2, school_info_attributes: school_attributes)` (note this returns an array of the created teachers).